internal/flags/bool.go

```diff
@@ -0,0 +1,99 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package flags
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/spf13/pflag"
+)
+
+const negPrefix = "no-"
+
+// AddNegationToBoolFlags adds the given flag in both `--foo` and `--no-foo` variants.
+// --no-foo is hidden from help output.
+func AddNegationToBoolFlags(f *pflag.FlagSet, p *bool, name string, value bool, usage string) {
+	negativeName := negPrefix + name
+
+	f.BoolVarP(p, name, "", value, usage)
+	f.Bool(negativeName, !value, "(negative) "+usage)
+
+	if value {
+		err := f.MarkHidden(name)
+		if err != nil {
+			panic(err)
+		}
+	} else {
+		err := f.MarkHidden(negativeName)
+		if err != nil {
+			panic(err)
+		}
+	}
```

Comment:
```suggestion
	if !value {
		name = negativeName
	}
	if err := f.MarkHidden(name); err != nil {
		panic(err)
	}
```
Wdyt of something like this?